jit-service/src/main/java/org/dcsa/jit/service/TimestampRoutingService.java

```diff
@@ -4,44 +4,142 @@
 import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.http.base.HttpOperationFailedException;
+import org.apache.camel.model.dataformat.JsonLibrary;
 import org.dcsa.jit.persistence.entity.MessageRoutingRule;
 import org.dcsa.jit.persistence.entity.OutboxMessage;
+import org.dcsa.jit.persistence.entity.TimestampNotificationDead;
 import org.dcsa.jit.persistence.repository.MessageRoutingRuleRepository;
 import org.dcsa.jit.persistence.repository.OutboxMessageRepository;
 import org.dcsa.jit.transferobjects.TimestampTO;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.time.OffsetDateTime;
+import java.util.HashMap;
 import java.util.List;
 
 @Slf4j
 @RequiredArgsConstructor
 @Service
-public class TimestampRoutingService {
+public class TimestampRoutingService extends RouteBuilder {
+
   private final MessageRoutingRuleRepository messageRoutingRuleRepository;
   private final OutboxMessageRepository outboxMessageRepository;
   private final ObjectMapper objectMapper;
 
+  @Override
+  public void configure() throws Exception {
+    pollOutstandingRoutingMessages();
+    emitMessage();
+    tokenOIDC();
+    timestampNotification();
+  }
+
+  void pollOutstandingRoutingMessages() {
+    from("{{camel.route.outbox-msg}}")
+        .routeId("outbox-message-poll")
+        .errorHandler(noErrorHandler())
+        .to("{{camel.route.emit-message}}");
+  }
+
+  void emitMessage() {
+    from("{{camel.route.emit-message}}")
+      .routeId("emit-msg")
+      .errorHandler(noErrorHandler())
+      .log("processing msg: ${body.payload}")
+      .setProperty("outboxMessage", simple("${body}"))
+      .choice()
+        // outbox_message has a not null constraint on message_routing_rule_id, so it is always
+        // present
+        // hence a check for null is not required
+        .when(simple("${body.messageRoutingRule.loginType} == 'OIDC'"))
+          .setProperty("credentials", simple("${body.messageRoutingRule.loginInformation}"))
+          .to("{{camel.route.oidc}}")
+        .otherwise()
+          .log("Unhandled loginType detected ${body.messageRoutingRule.loginType}")
+          .stop()
+      .end()
+      .to("{{camel.route.timestamp-notify}}");
```

Comment:
verify which HTTP headers are sent Camel has a tendency to send a lot more headers then one would expect. If there are indeed more headers then expected use the .removeHeaders EIP: https://camel.apache.org/components/3.18.x/eips/removeHeaders-eip.html

Reply:
I did check, no unintended headers are being passed.